SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,412 @@
+#include "LoadBalancing/LayeredLBStrategy.h"
+#include "Tests/AutomationCommon.h"
+#include "Tests/AutomationEditorCommon.h"
+#include "Tests/TestDefinitions.h"
+#include "Engine/Engine.h"
+#include "GameFramework/GameStateBase.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "LoadBalancing/GridBasedLBStrategy.h"
+#include <vector>
+#include <utility>
+#include "SpatialGDKTests/SpatialGDK/LoadBalancing/GridBasedLBStrategy/TestGridBasedLBStrategy.h"
+#include <string>
+#include "TestLayeredLBStrategy.h"
+#include "GameFramework/DefaultPawn.h"
+#include "Misc/Optional.h"
```

Comment:
Our preference for imports is to group imports based on their source and then alphabetized within their group. The groups ordering would generally be: GDK then engine imports then worker SDK imports then c language include:

In this case, I think yours would be:

#include "EngineClasses/SpatialWorldSettings.h"
#include "LoadBalancing/GridBasedLBStrategy.h"
#include "LoadBalancing/LayeredLBStrategy.h"
#include "SpatialGDKTests/SpatialGDK/LoadBalancing/GridBasedLBStrategy/TestGridBasedLBStrategy.h"
#include "TestLayeredLBStrategy.h"

#include "Engine/Engine.h"
#include "GameFramework/DefaultPawn.h"
#include "GameFramework/GameStateBase.h"
#include "Misc/Optional.h"
#include "Tests/AutomationCommon.h"
#include "Tests/AutomationEditorCommon.h"
#include "Tests/TestDefinitions.h"

#include <string>
#include <utility>
#include <vector>

Not sure if the 4th include in my list can be simplified

Reply:
Also this list is a first glance, might've got a couple wrong

Reply:
Writing up this guideline is something I've wanted to do for a while but here's a quick summary: the reason is to do with preventing you omitting necessary includes. If you follow these include ordering rules you're more likely to get compiler errors telling you about missing includes. If you don't, you risk getting away with it for now, but then cause quite ambiguous errors for others in future. 

Alex gave me a good example case of this:

Imagine there are the following files:
Engine/A.h
GDK/B.h
GDK/C.h
GDK/New.h

Now imagine GDK/C.h includes Engine/A.h and then GDK/B.h. GDK/B.h relies on some Engine/A.h stuff, but doesn't include Engine/A.h - still everything works fine! (because of order of includes in GDK/C.h). Then GDK/New.h includes GDK/B.h - and the compilation breaks.
So, if GDK/C.h included GDK/B.h and then Engine/A.h - the compilation would have broken earlier, and tell GDK/B.h misses the Engine/A.h include.

Reply:
yep fair enough, done